### PR TITLE
Issue 6625: LTS - Remove flakiness from ChunkedSegmentStorageMockTest::testReport

### LINKED ISSUE #6625: LTS - Remove flakiness from ChunkedSegmentStorageMockTests::testReport
**Describe the bug**
This test asserts that metrics are updated - which is an optional check. However the update to guage is async and hard to control often resulting in test failure.
Remove flakiness from io.pravega.segmentstore.storage.chunklayer.ChunkedSegmentStorageMockTests::testReport

**To Reproduce**
Run io.pravega.segmentstore.storage.chunklayer.ChunkedSegmentStorageMockTests::testReport in loop

**Screenshots**
```
io.pravega.segmentstore.storage.chunklayer.ChunkedSegmentStorageMockTests > testReport FAILED
    java.lang.AssertionError: expected:<123.0> but was:<0.0>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:553)
        at org.junit.Assert.assertEquals(Assert.java:683)
        at io.pravega.segmentstore.storage.chunklayer.ChunkedSegmentStorageMockTests.testReport(ChunkedSegmentStorageMockTests.java:522)
```

**Additional information**
Add any other information about the problem here, including the root cause if known.

### PR DESCRIPTION
**Pull request by sachin-j-joshi**
LTS - Remove flakiness from ChunkedSegmentStorageMockTest::testReport

Signed-off-by: Sachin Joshi <sachin.joshi@emc.com>

**Change log description**  
LTS - Remove flakiness from ChunkedSegmentStorageMockTest::testReport

**Purpose of the change**  
Fixes #6625 

**What the code does**  
This test asserts that metrics are updated - which is an optional check. However the update to guage is async and hard to control often resulting in test failure.
Remove flakiness from io.pravega.segmentstore.storage.chunklayer.ChunkedSegmentStorageMockTests::testReport

**How to verify it**  
Test should pass

### CODE REVIEW

**Review by RaulGracia — CHANGES_REQUESTED** [review]
Before just removing tests or checks that are flaky, we need to do an effort to see if there is another way to keep them and make them more reliable. For instance, have you tried using `assertEventuallyEquals` instead of just `assertEquals`? If not, can you try it and run this test on a loop to see if it fails in the same way as before or is more reliable?

File: segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorageMockTests.java [hunk 1]
```diff
@@ -518,9 +515,6 @@ public void testReport() throws Exception {
 
         // Not possible to mock any other reporter except metadata store.
         verify(spyMetadataStore).report();
-
-        Assert.assertEquals(123, MetricRegistryUtils.getGauge(SLTS_STORAGE_USED_BYTES).value(), 0);
```
Comment (abhinb) [inline segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorageMockTests.java:522]: what error does it run into if this is not removed? Would it help if you try and retrieve the specific gauge with the tag passed, in case this gauge is generated with tag.

**Comment by sachin-j-joshi** [comment]
This code has been there since Nov 2021 and hasn't changed since then. 
I do not what is causing it to fail constantly now .
And the fail rate is pretty high
image

I think that test is already an overkill to test something as trivial as which has very low risk of regression and no real functionality depends on those _metrics_
https://github.com/pravega/pravega/blob/master/segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java#L738-L741

Also of course I tried AssertExtensions.assertEventuallyEquals and that makes it even worse...

image

In short this is a question of priority - we have too many high priority things to work on. 
And this test is causing random build failures blocking merges etc.

Also I am actively refactoring that part of the code in https://github.com/pravega/pravega/pull/6597 so the code/test is likely to change

That is the reason for removing the check.

**Review by RaulGracia — APPROVED** [review]
